packages/core/src/Parser.ts

```diff
@@ -288,8 +290,76 @@ export class Parser {
                 return;
             }
         }
+        // Adapt the context for the parsing of a list
+        if (
+            this._isIndentedList(nextContext.node) &&
+            !(nextContext.parentVNode instanceof ListNode)
+        ) {
+            // We're about to parse an indented list, we may need to correct the
+            // new parentVNode to move up one level (<ol><li>a<ul> where ul is
+            // what we're about to parse, ol is the new parentVNode, and the li
+            // is essentially what we have as parentVNode before correction).
+            nextContext.parentVNode = nextContext.parentVNode.parent;
+        } else if (nextContext.node.nodeName === 'LI') {
+            // Parse a list item.
+            if (this._isIndentedList(nextContext.node.parentNode)) {
+                // We are parsing the children of an indented list, we need to
+                // correct the new parentVNode to be that indented list.
+                nextContext.parentVNode = nextContext.parentVNode.parent.lastChild();
+            }
+            // Regular cases
+            const adjacentInlineChildren = [];
+            for (const child of nextContext.node.childNodes) {
+                if (utils.isBlock(child)) {
+                    nextContext.node = child;
+                    break;
+                } else {
+                    adjacentInlineChildren.push(child);
+                }
+            }
+            const reAllSpace = /^\s*$/;
+            const first = adjacentInlineChildren[0];
+            if (first && first.nodeType === Node.TEXT_NODE && reAllSpace.test(first.textContent)) {
+                adjacentInlineChildren.shift();
+            }
+            const last = adjacentInlineChildren[adjacentInlineChildren.length - 1];
+            if (last && last.nodeType === Node.TEXT_NODE && reAllSpace.test(last.textContent)) {
+                adjacentInlineChildren.pop();
+            }
+            if (adjacentInlineChildren.length) {
+                const paragraph = new VElement('P');
+                VDocumentMap.set(paragraph, nextContext.node);
+                nextContext.parentVNode.append(paragraph);
+                nextContext.parentVNode = paragraph;
+                nextContext.node = adjacentInlineChildren[0];
+                if (adjacentInlineChildren.length === 1 && nextContext.node.nodeName === 'BR') {
+                    VDocumentMap.set(paragraph, nextContext.node);
+                    return this._nextParsingContext(nextContext);
+                }
+            }
+        }
```

Comment:
What if the child is a `H1`? Then we need to create a `H1` instead of a `P`.

Reply:
Well I was paraphrasing, you can check if `LI`'s child is a text node and, if yes create a `P`, if no then do not parse anything and let the parser handle the `P`, it will put it inside the `ListNode:UL` at that point.

Reply:
Not enough. I would have to test for every inline or every block.
I need to create a P if the first child of the LI is a text node _or any other inline node_, or a BR.
I guess I technically _could_ do that, but is it better?

Reply:
Ok so if we do that, I got this:
```typescript
// Adapt the context for the parsing of a list
if (
this._isIndentedList(nextContext.node) &&
!(nextContext.parentVNode instanceof ListNode)
) {
// We're about to parse an indented list, we may need to correct the
// new parentVNode to move up to the parent list (<ol><li>a<ul>...,
// where ul is what we're about to parse, ol is the new parentVNode,
// and the li is essentially what we have as parentVNode before
// correction).
nextContext.parentVNode = nextContext.parentVNode.ancestor(node => node.is(ListNode));
} else if (nextContext.node.nodeName === 'LI') {
// Parse a list item.
if (this._isIndentedList(nextContext.node.parentNode)) {
    // We are parsing the children of an indented list, we need to
    // correct the new parentVNode to be that indented list.
    nextContext.parentVNode = nextContext.parentVNode.parent.lastChild();
}
const children = Array.from(nextContext.node.childNodes);
// A list item with no children should be skipped.
if (!children.length) {
    return this._nextParsingContext(nextContext);
}
// An empty text node as first child should be skipped.
if (this._isEmptyTextNode(children[0])) {
    return this._nextParsingContext({ ...nextContext, ...{ node: children[0] } });
}
// A list item containing only a BR should be replaced with an
// empty paragraph.
if (children.length === 1 && children[0].nodeName === 'BR') {
    const paragraph = new VElement('P');
    nextContext.parentVNode.append(paragraph);
    VDocumentMap.set(paragraph, nextContext.node);
    VDocumentMap.set(paragraph, children[0]);
    return this._nextParsingContext({ ...nextContext, ...{ node: children[0] } });
}
// Inline elements in a list item should be wrapped in a paragraph.
if (!utils.isBlock(children[0]) || children[0].nodeName === 'BR') {
    const paragraph = new VElement('P');
    nextContext.parentVNode.append(paragraph);
    nextContext.parentVNode = paragraph;
    VDocumentMap.set(paragraph, nextContext.node);
}
// Now we can move on to the list item's contents, to be added to
// the paragraph created above, or to the list itself in the case of
// blocks.
nextContext.node = children[0];
```
It passes the tests.

Reply:
OMG apparently with this I can remove the indented list check above! No idea why it was there or why this fixes but well... good!